Pre-packaged Linux binary thinks user is on App Engine

The Google Sheets preprocessor seems to think I'm running on App Engine for some reason. I believe this is happening in https://github.com/grow/grow/blob/master/grow/common/oauth.py, where we determine if we're running on App Engine or not based on whether the import fails. Can we switch to a more deterministic way of checking if the user is on App Engine?

```
[15:30:31] Compiled: en_US (280/332)
Attempting refresh to obtain initial access_token
Traceback (most recent call last):
  File "bin/grow", line 76, in <module>
  File "site-packages/click/core.py", line 722, in __call__
  File "site-packages/click/core.py", line 697, in main
  File "site-packages/click/core.py", line 1066, in invoke
  File "site-packages/click/core.py", line 895, in invoke
  File "site-packages/click/core.py", line 535, in invoke
  File "grow/commands/subcommands/preprocess.py", line 42, in preprocess
  File "grow/pods/pods.py", line 781, in preprocess
  File "grow/preprocessors/google_drive.py", line 51, in run
  File "grow/preprocessors/google_drive.py", line 338, in execute
  File "grow/preprocessors/google_drive.py", line 191, in download
  File "grow/preprocessors/google_drive.py", line 47, in create_service
  File "site-packages/oauth2client/util.py", line 137, in positional_wrapper
  File "site-packages/googleapiclient/discovery.py", line 198, in build
  File "site-packages/oauth2client/transport.py", line 153, in new_request
  File "site-packages/oauth2client/contrib/appengine.py", line 178, in _refresh
  File "google/appengine/api/app_identity/app_identity.py", line 600, in get_access_token
  File "google/appengine/api/memcache/__init__.py", line 562, in get
  File "google/appengine/api/memcache/__init__.py", line 619, in get_multi_async
  File "google/appengine/api/memcache/__init__.py", line 385, in _make_async_call
  File "google/appengine/api/memcache/__init__.py", line 297, in create_rpc
  File "google/appengine/api/apiproxy_stub_map.py", line 414, in __init__
  File "google/appengine/api/apiproxy_stub_map.py", line 68, in CreateRPC
AssertionError: No api proxy found for service "memcache"
```

--- a/grow/common/oauth.py
+++ b/grow/common/oauth.py
@@ -34,7 +34,7 @@ DEFAULT_AUTH_KEY_FILE = 'auth-key.json'
 
 def get_storage(key, username):
     """Returns the Storage class compatible with the current environment."""
-    if appengine:
+    if appengine and utils.is_appengine():
         return appengine.StorageByKeyName(
             appengine.CredentialsModel, username, 'credentials')
     key = utils.slugify(key)

--- a/grow/common/utils.py
+++ b/grow/common/utils.py
@@ -33,6 +33,7 @@ except ImportError:
     from yaml import Loader as yaml_Loader
 
 
+APPENGINE_SERVER_PREFIXES = ('Development/', 'Google App Engine/')
 LOCALIZED_KEY_REGEX = re.compile('(.*)@([^@]+)$')
 SENTINEL = object()
 SLUG_REGEX = re.compile(r'[\t !"#$%&\'()*\-/<=>?@\[\\\]^_`{|},.]+')
@@ -59,10 +60,9 @@ def is_packaged_app():
 
 def is_appengine():
     """Returns whether the environment is Google App Engine."""
-    if 'SERVER_SOFTWARE' in os.environ:
-        # https://cloud.google.com/appengine/docs/standard/python/how-requests-are-handled
-        return os.environ['SERVER_SOFTWARE'].startswith(('Development/', 'Google App Engine/'))
-    return False
+    # https://cloud.google.com/appengine/docs/standard/python/how-requests-are-handled
+    return os.getenv('SERVER_SOFTWARE', '').startswith(
+        APPENGINE_SERVER_PREFIXES)
 
 
 def get_git():

